pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SingleSnapshotAbortedTxnProcessorImpl.java

```diff
@@ -87,7 +90,16 @@ public CompletableFuture<PositionImpl> recoverFromSnapshot() {
                     PositionImpl startReadCursorPosition = null;
                     try {
                         while (reader.hasMoreEvents()) {
-                            Message<TransactionBufferSnapshot> message = reader.readNext();
+                            Message<TransactionBufferSnapshot> message = reader.readNext(2, TimeUnit.SECONDS);
```

Comment:
1. 
```
try {
    Message<TransactionBufferSnapshot> message = reader.readNextAsync().get(30, TimeUnit.SECONDS);
} catch (Exception ex) {
    Throwable t = FutureUtil.unwrapCompletionException(ex);
    log.error("[{}] Transaction buffer recover fail when read "
    + "transactionBufferSnapshot!", topic.getName(), t);
    closeReader(reader);
    return FutureUtil.failedFuture(t);
}
```
better to use this logic, topicTransactionBuffer use getTransactionExecutorProvider thread recover, so we don't need to care about the deadlock problem. and don't need to add any new API.
2. it's better to use timeout client `operationTimeoutMs`, 2 seconds is too short

Reply:
Good suggestion, already fixed

Reply:
How to try recovery again if implemented in this way?